web/modules/custom/form_api/src/Form/UserForm.php

```diff
@@ -0,0 +1,243 @@
+<?php
+
+namespace Drupal\form_api\Form;
+
+use Drupal\Component\Utility\EmailValidatorInterface;
+use Drupal\Core\Ajax\AddCssCommand;
+use Drupal\Core\Ajax\AjaxResponse;
+use Drupal\Core\Ajax\HtmlCommand;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides user form for user information.
+ *
+ * @internal
+ */
+class UserForm extends ConfigFormBase {
+
+  /**
+   * Constant used to set or get state.
+   *
+   * @var string
+   */
+  const FORM_API_CONFIG_PAGE = 'form_api_config_page:values';
+
+  /**
+   * Validates the email.
+   *
+   * @var \Drupal\Component\Utility\EmailValidatorInterface
+   */
+  protected $emailValidator;
+  /**
+   * Gets value from config.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  protected $configFactory;
+
+  /**
+   * Contructs the object of the class.
+   *
+   * @param \Drupal\Component\Utility\EmailValidatorInterface $email_validator
+   *   The email validator.
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   Object of config factory to get and set values in form.
+   */
+  public function __construct(EmailValidatorInterface $email_validator, ConfigFactoryInterface $config_factory) {
+    $this->emailValidator = $email_validator;
+    $this->configFactory = $config_factory;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('email.validator'),
+      $container->get('config.factory')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'form_api_config_page';
+  }
+
+  /**
+   * Gets editable config file name.
+   */
+  public function getEditableConfigNames() {
+    return ['form_api.settings'];
+  }
+
+  /**
+   * Form constructor.
+   *
+   * Display a tree of all the form elements of user form.
+   *
+   * @param array $form
+   *   An associative array containing the structure of the form.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The current state of the form.
+   *
+   * @return array
+   *   The form structure.
```

Comment:
Do we need such extensive documentation for these inherited functions?

Reply:
Fixed.